### PR TITLE
Use shared CD workflow

### PR DESCRIPTION
**Pull request by jglick**
Testing https://github.com/jenkins-infra/github-reusable-workflows/pull/1. See https://github.com/jenkinsci/gitlab-api-plugin/commit/b8c272862e86356500ac47c0139b8a09e0f99d46

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -1 +0,0 @@
-_extends: .github
```
Comment (jglick) [inline .github/release-drafter.yml:1]: Supposedly no longer necessary.

File: .mvn/extensions.xml [hunk 2]
```diff
@@ -2,6 +2,6 @@
   <extension>
     <groupId>io.jenkins.tools.incrementals</groupId>
     <artifactId>git-changelist-maven-extension</artifactId>
-    <version>1.1</version>
```
Comment (jglick) [inline .mvn/extensions.xml:5]: Old!

File: pom.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>4.17</version>
```
Comment (jglick) [inline pom.xml:7]: Also old!

File: pom.xml [hunk 4]
```diff
@@ -13,9 +13,8 @@
     <version>${changelist}</version>
     <packaging>hpi</packaging>
     <properties>
-        <changelist>1.2-SNAPSHOT</changelist>
```
Comment (jglick) [inline pom.xml:16]: incorrect

File: pom.xml [hunk 5]
```diff
@@ -13,9 +13,8 @@
...
+        <changelist>999999-SNAPSHOT</changelist>
         <jenkins.version>2.235.1</jenkins.version>
-        <java.level>8</java.level>
```
Comment (jglick) [inline pom.xml:18]: Obsolete as of parent 4.40.

**Review by timja — APPROVED** [review]

**Comment by jglick** [comment]
Is it OK to merge this despite pulling in an unmerged branch?

**Comment by timja** [comment]
yes this is a test plugin do whatever you want, it's not installed anywhere

**Comment by jglick** [comment]
Hmm, did not work. https://github.com/jenkinsci/jenkins-infra-test-plugin/runs/6563842984?check_suite_focus=true found no interesting changes despite https://github.com/jenkinsci/jenkins-infra-test-plugin/releases/tag/untagged-dfca8f2b610ab16de339 looking right. I have seen a similar problem in another plugin recently (not using the shared workflow of course). I wonder if GH has some sort of time lag in releases? If so, we might need to introduce an artificial delay into the workflow. @jetersen

**Pull request merged**